models/lobby/configs.go

```diff
@@ -0,0 +1,270 @@
+package lobby
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/TeamPlayTF/Server/config"
+)
+
+const (
+	ConfigsPath = "/configs/"
+	ConfigsFile = "configs.json"
+	MapsFile    = "maps.json"
+)
+
+var ConfigsData *ConfigData
+
+// MapsData holds the map + config list from maps.json
+//
+// think about it like this:
+// -> MapsData["MAP NAME"].V6.Ugc -> would return the ugc's 6v6 cfg for that map
+// -> MapsData[string].GameMode.LeagueConfig
+var MapsData map[string]GameMode
+
+type LeagueConfig struct {
+	Ugc   string `json:"ugc"`
+	Etf2l string `json:"etf2l"`
+}
+
+type GameMode struct {
+	V6 *LeagueConfig `json:"6v6"`
+	V9 *LeagueConfig `json:"9v9"`
+}
+
+// league configs from configs.json
+type ConfigData struct {
+	Etf2l struct {
+		V6 []string `json:"6v6"`
+		V9 []string `json:"9v9"`
+	} `json:"etf2l"`
+
+	Ugc struct {
+		V6 []string `json:"6v6"`
+		V9 []string `json:"9v9"`
+	} `json:"ugc"`
+}
+
+// configs.json
+type ServerConfig struct {
+	Name   string    // example: HL_stopwatch
+	League string    // ugc, etf2l...
+	Type   LobbyType // 9v9, 6v6...
+	Data   string    // config file's text
+	Map    string
+}
+
+// maps.json
+type MapConfig struct {
+	Name   string        // map name
+	Config *ServerConfig // config info (name, league, type and data)
+}
+
+func InitConfigs() {
+	// configs
+	fmt.Println("[Configs.Init] Loading server configs...")
+	cfgFile, cfgErr := ioutil.ReadFile(config.Constants.StaticFileLocation + ConfigsPath + ConfigsFile)
+
+	if cfgErr == nil {
+		json.Unmarshal(cfgFile, &ConfigsData)
+		fmt.Println("[Configs.Init] Server configs loaded!")
+
+	} else {
+		fmt.Println("[Configs.Init] ERROR while trying to load server configs!")
+		log.Fatal(cfgErr)
+	}
+
+	// maps
+	fmt.Println("[Configs.Init] Loading maps configs...")
+	mapFile, mapErr := ioutil.ReadFile(config.Constants.StaticFileLocation + ConfigsPath + MapsFile)
+
+	if mapErr == nil {
+		json.Unmarshal(mapFile, &MapsData)
+		fmt.Println("[Configs.Init] Maps configs loaded!")
+
+	} else {
+		fmt.Println("[Configs.Init] ERROR while trying to load maps configs!")
+		log.Fatal(mapErr)
+	}
+}
+
+func NewServerConfig() *ServerConfig {
+	return new(ServerConfig)
+}
+
+func (c *ServerConfig) Get() (string, error) {
+	if c.League == "" {
+		return "", errors.New("[Configs.Get]: No league specified!")
+	}
+
+	if c.Type != LobbyTypeSixes && c.Type != LobbyTypeHighlander {
+		return "", errors.New("[Configs.Get]: The type you specified doesn't exists!")
+	}
+
+	if c.Name == "" {
+		configName, configNameErr := c.GetMapConfig(c.Map)
+
+		// if happens, rip
+		if configNameErr != nil {
+			log.Fatal(configNameErr)
```

Comment:
should you really log.Fatal in cases like this? You should probably just stop, return an error and let the controller send an error response